app/src/main/java/com/example/taskyy/domain/repository/AuthRepository.kt

```diff
@@ -1,8 +1,9 @@
 package com.example.taskyy.domain.repository
 
-import com.example.taskyy.data.data_access_objects.UserDao
-import com.example.taskyy.data.room_entity.UserEntity
+import com.example.taskyy.data.local.data_access_objects.UserDao
+import com.example.taskyy.data.local.room_entity.UserEntity
 
 interface AuthRepository {
-    fun registerUser(userEntity: UserEntity, userDao: UserDao)
+    fun addUserToDatabase(userEntity: UserEntity, userDao: UserDao)
+    fun registerUser(userEntity: UserEntity)
 }
```

Comment:
Here, you're breaking the principle that the domain layer should be isolated. Isolation here means that domain shouldn't know about any implementation details (**how** something is implemented, e.g. which library is used for HTTP calls, how data is stored in a DB or how your request/response classes look like).
In this interface which sits in domain (that itself is correct), you're accessing classes from data however which is not allowed. The idea behind this isolation is that frequently changing code (implementation details) doesn't cause changes in less frequently changing code (business logic). If you break this principle like here, for example if you'd change your `UserEntity`, this would directly cause a change here in the repo. 

Instead, only pass data to the repository that's needed there. For the `registerUser` function for example, all the repository needs to know is the full name, email and password the user wants to register with. The DAO you can just inject in the repository then:
```
class AuthRepositoryImpl(
    private val userDao: UserDao,
    private val taskyApi: TaskyyApi
): AuthRepository {
    // ...
}
```
And provide the repository like this in your AppModule:
```
@Provides
@Singleton
fun provideAuthRepository(userDao: UserDao, api: TaskyyApi): AuthRepository {
    return AuthRepositoryImpl(userDao, api)
}
```
Now that Dagger knows how to create such a repository reference, you can inject this repository wherever you want (typically in a ViewModel):
```
@HiltViewModel
class RegisterViewModel @Inject constructor(
    private val repository: AuthRepository
): ViewModel() { ... }
```
I hope this long comment wasn't too confusing. If it was, please ask 😄